crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1014,33 +1018,25 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
             .map(|((id, _, _), owner)| (*owner, *id));
         write_batch =
             write_batch.delete_batch(&self.perpetual_tables.owner_index, owners_to_delete)?;
-        let mutated_objects = effects
-            .mutated
-            .iter()
-            .map(|(r, _)| r)
-            .chain(effects.deleted.iter())
-            .chain(effects.wrapped.iter())
-            .map(|(id, version, _)| {
-                ObjectKey(
-                    *id,
-                    version
-                        .decrement()
-                        .expect("version revert should never fail"),
-                )
-            });
-        let old_objects = self
+
+        let old_mutable_inputs = self
             .perpetual_tables
             .objects
-            .multi_get(mutated_objects)?
+            .multi_get(input_object_keys(&effects, &certificate)?)?
```

Comment:
Having read the rest so I am not sure, but does this handle well for the dynamic child objects that's not explicitly part of the input?

Reply:
Yep, this is a good point, it does not.  The previous implementation of this introduced a field in effects to store , it was pointed out that it isn't currently necessary to do that, but that is now no longer true because of dynamic fields.

This effects #5748 rather than this diff, but I think it's an argument for rebasing this change first, and then tackling the increment and  decrement cases of lamport timestamps together (because in order to have access to the old versions, we need to move the increments to some nice, central place, which I'm in the process of doing).